backend/src/auth_handlers.py

```diff
@@ -1,131 +1,63 @@
-import server
 import requests
-from flask import send_file, request, jsonify
-from datetime import datetime, timezone, timedelta
-from jwt import encode, decode, ExpiredSignatureError, exceptions as jwtExceptions
-import logging
-from models import User
+from flask import request, jsonify, Blueprint
 
+import app
 
-def register_new_user():
+auth_module = Blueprint('auth',__name__,url_prefix='/auth')
+
+@auth_module.route('/register', methods=['POST'])
+def register():
     """registers a new user in the DB, returns JWT Token
     :return: flask response object
     """
-    if request.get_json().get("user") is None:
-        return server.err_out(401, "user is a required field")
     # get user info from front end
     try:
-        user_data = request.get_json().get("user")
-        userd = User(user_data)
-    except KeyError as err:
-        return server.err_out(401, "'{}' is a required field".format(str(err)))
-
-    # check if user already exists
-    if server.db.username_already_exists(userd.username):
-        return server.err_out(
-            409, "username {} already exists".format(userd.username))
-
+        user_data = request.get_json() #dictionary of input
+        username = user_data.get('user_name')
+        password = user_data.get('password')
+        role = user_data.get('role')
+        is_admin = bool(user_data.get('is_admin'))
+        if username and password and role and is_admin:
+            created_user = app.services.user_service().add_user(username,password,role,is_admin)
+            return jsonify({'jwt': app.services.auth_service().encode_jwt(created_user)})
+        else:
+            return jsonify(code=400, error='Request is invalid'), 400
+    except KeyError:
+        return jsonify(code=400, error='Request is invalid'), 400
```

Comment:
errors should be logged here for debugging if a request returns a non-200 response

Reply:
Yep, done